deploy-clojars/action.yml

```diff
@@ -0,0 +1,89 @@
+name: Clojars Deployment
+
+description: Build JAR and Deploy to Clojars
+
+inputs:
+  group-id:
+    description: The Clojars Group ID. Defaults to 'com.yetanalytics'.
+    required: false
+    default: 'com.yetanalytics'
+
+  artifact-id:
+    description: The Clojars Artifact ID (i.e. the name of the lib itself)
+    required: true
+
+  version:
+    description: >
+      The version string. Defaults to the Git branch or tag associated with
+      the event that triggered this action. NOTE: If the tag is prefixed
+      (e.g. 'v1.0.0'), you should remove the prefix, then pass in the
+      truncated tag as the arg (e.g. pass in `${GITHUB_REF#refs\/tags\/v}`).
+    required: false
+    default: ${{ github.ref_name }}
+
+  src-dirs:
+    description: >
+      A quoted array of the source directories. Defaults to '["src/main"]'.
+    required: false
+    default: '["src/main"]'
+
+  resource-dirs:
+    description: >
+      A quoted array of the resource directories. Defaults to '["resources"]'.
+      Pass '[]' if no resource directories exist.
+    required: false
+    default: '["resources"]'
+
+  # GitHub Secrets
+
+  clojars-username:
+    description: The Clojars username (should be stored as a GitHub secret).
+    required: true
+
+  clojars-password:
+    description: The Clojars password (should be an access token stored as a GitHub secret).
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Add aliases
+      shell: bash
+      run: >
+        clojure -Sdeps
+        '{:aliases
+          {:build
+           {:replace-deps {io.github.seancorfield/build-clj {:git/tag "v0.6.3" :git/sha "9b8e09b"}}
+            :ns-default build}}}'
+
+    - name: Build JAR file
+      shell: bash
+      run: >
+        clojure -X:build jar
+        :lib ${{ inputs.group-id }}/${{ inputs.artifact-id }}
+        :version '"${{ inputs.version }}"'
+        :src-dirs '${{ inputs.src-dirs }}'
+        :resource-dirs '${{ inputs.resource-dirs }}'
+        :scm '{
+          :url "https://github.com/${{ github.repository }}"
```

Comment:
wonder why this wouldn't get cut off tho, weird

Reply:
Any ideas?

Reply:
That's odd, it _does_ make it newlines https://github.com/yetanalytics/actions/runs/4653587786?check_suite_focus=true#step:4:21

Reply:
![image](https://user-images.githubusercontent.com/782173/147607907-166012f5-1d91-4760-8a27-8030bd0839dc.png)
creepy

Reply:
It happens when you indent the _last_ line of the `'{:aliases ...}'` map.

Reply:
<img width="1784" alt="Screen Shot 2021-12-28 at 4 23 58 PM" src="https://user-images.githubusercontent.com/32022576/147608127-2071861b-ed4a-413a-b82d-79acd9bf6732.png">

Reply:
^ Nothing happens if you indent the _penultimate_ line.

Reply:
Then why does it render on multiple lines in the github actions log? Is that just a bug?

Reply:
At any rate, it appears to work and run. I'm gonna chalk this up to more syntax more (sugary) problems.

Reply:
Oh in this case it's because I'm passing the `'{:aliases ...}'` string as the output of a step, so all the linebreaks get removed. Not so for `:scm '{...}'`, where the linebreaks are preserved since it's a hardcoded string.

Reply:
Ahh, that makes sense to me.